bigquery/tests/integration.sh

```diff
@@ -32,87 +33,152 @@ waitForClusterHealthy() {
+    exit 1
+  fi
+
+  # Write the mappings out to files in order to diff.
+  curl -s 'http://localhost:9200/framingham_heart_study_teaching_dataset/_mappings?pretty' | jq -rS '.' > 'tests/framingham_heart_study_teaching_dataset_mappings.json'
+  DIFF=$(diff tests/framingham_heart_study_teaching_dataset_mappings_golden.json tests/framingham_heart_study_teaching_dataset_mappings.json)
+  if [ "$DIFF" != "" ]; then
+    echo "Mappings do not match golden json file, diff:"
+    echo "$DIFF"
+    exit 1
+  fi
+
+  # Write the fields indexes out to files in order to diff.
+  curl -s 'http://localhost:9200/framingham_heart_study_teaching_dataset_fields/_search?size=200' | jq -rS '.hits.hits' > 'tests/framingham_heart_study_teaching_dataset_fields.json'
+  DIFF=$(diff tests/framingham_heart_study_teaching_dataset_fields_golden.json tests/framingham_heart_study_teaching_dataset_fields.json)
+  if [ "$DIFF" != "" ]; then
+    echo "Fields index does not match golden json file, diff:"
+    echo "$DIFF"
+    exit 1
+  fi
+}
+readonly -f validate_framingham_heart_study_teaching_dataset
+
+
+function validate_1000_genomes() {
+  # Validate the correct number of documents were indexed.
+  DOC_COUNT_GENOMES=$(curl -s 'http://localhost:9200/1000_genomes/_search' | jq -r '.hits.total')
+  if [ "$DOC_COUNT_GENOMES" != "3500" ]; then
+    echo "Number of documents is incorrect, expected 3500, got $DOC_COUNT_GENOMES"
+    exit 1
+  fi
+
+  # Write the indexes out to a file in order to diff.
+  curl -s 'http://localhost:9200/1000_genomes/type/HG02924' | jq -rS '._source' > 'tests/1000_genomes.json'
+  DIFF=$(diff tests/1000_genomes_golden.json tests/1000_genomes.json)
+  if [ "$DIFF" != "" ]; then
+    echo "Index does not match golden json file, diff:"
+    echo "$DIFF"
+    exit 1
+  fi
+
+  # Write the mappings out to files in order to diff.
+  curl -s 'http://localhost:9200/1000_genomes/_mappings' | jq -rS '.' > 'tests/1000_genomes_mappings.json'
+  DIFF=$(diff tests/1000_genomes_mappings_golden.json tests/1000_genomes_mappings.json)
+  if [ "$DIFF" != "" ]; then
+    echo "Mappings do not match golden json file, diff:"
+    echo "$DIFF"
+    exit 1
+  fi
+
+  # Write the fields indexes out to files in order to diff.
+  curl -s 'http://localhost:9200/1000_genomes_fields/_search?size=200' | jq -rS '.hits.hits' > 'tests/1000_genomes_fields.json'
+  DIFF=$(diff tests/1000_genomes_fields_golden.json tests/1000_genomes_fields.json)
+  if [ "$DIFF" != "" ]; then
+    echo "Fields index does not match golden json file, diff:"
+    echo "$DIFF"
+    exit 1
+  fi
+}
+readonly -f validate_1000_genomes
+
+
+function test_columns_to_ignore() {
+  # Index a dataset that has columns named chr_1_vcf, chr_2_vcf, etc
+  # In the config files, explicitly mark that chr_3_vcf as ignored
+  # Search should match for chr_1_vcf, but not chr_3_vcf
+  echo "Running test_columns_to_ignore"
+
+  DATASET_CONFIG_DIR=tests/columns_to_ignore_test docker-compose up --build indexer
+
+  sleep 5
```

Comment:
opt - In general it's better to wait for something, rather than sleep. If the thing is already ready, you don't waste 5 seconds.

If the sleep is unavoidable, add a comment.